src/pymediainfo/__init__.py

```diff
@@ -350,22 +349,61 @@ def can_parse(cls, library_file: Optional[str] = None) -> bool:
         except Exception:  # pylint: disable=broad-except
             return False
 
+    @overload
+    @classmethod
+    def parse(
+        # pylint: disable=too-many-arguments, too-many-locals
+        # pylint: disable=too-many-branches, too-many-statements
+        cls,
+        filename: Any,
+        *,
+        library_file: str | None = None,
+        cover_data: bool = False,
+        encoding_errors: str = "strict",
+        parse_speed: float = 0.5,
+        full: bool = True,
+        legacy_stream_display: bool = False,
+        mediainfo_options: dict[str, str] | None = None,
+        output: str,
```

Comment:
Can you add a comment above this explaining why you're using overloads? I have never used it before. IIUC, you do this because if output is None, we return a MediaInfo object, otherwise a string, and defining this precisely is better for testing?

Reply:
Exactly, it's for static type checkers (`mypy`). Otherwise they expect a possible `str` and complain when we use it as a `MediaInfo` instance.
It would be better to have 2 distinct functions eventually, `parse` and `parse_to_str`, for a better type stability.

Reply:
In hindsight, even the whole "guess if the filename is a URL" thing should have its own `parse_url` function I guess. Let's say that so far it works but it's not ideal.

Reply:
Yes, I saw that, but I didn't want to charge this PR.
Also the function to normalize paths is not needed I would say, as `os.fspath` is doing the job already.

Reply:
>  Also the function to normalize paths is not needed I would say, as os.fspath is doing the job already.

os.fspath was added to Python 3.6, I wonder if that's why they did that in cf2b7df1bdacd964c5e4d0483deff9c6c935e67e. I'll keep that in mind.